### PR TITLE
Fix contingency propagation

### PR DESCRIPTION
**Pull request by geofjamg**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**

Bug fix

**What is the current behavior?**

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [ ] No

**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

**Other information**:

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/network/impl/PropagatedContingency.java [hunk 1]
```diff
@@ -450,10 +460,10 @@ public Optional<LfContingency> toLfContingency(LfNetwork network) {
                         boolean otherSideConnected;
                         if (branch.getBus1() == busToLost) {
                             otherSideBus = branch.getBus2();
-                            otherSideConnected = branch.isConnectedSide2();
+                            otherSideConnected = branch.isConnectedSide2() && isConnectedAfterContingencySide2(branchesToOpen, branch);
```
Comment (rosiereflo) [inline src/main/java/com/powsybl/openloadflow/network/impl/PropagatedContingency.java:463]: Why don't we update the network g=following the contingency?

Comment (annetill) [reply src/main/java/com/powsybl/openloadflow/network/impl/PropagatedContingency.java:463]: This is done in the `apply` method of  `LfContingency`.

Comment (annetill) [reply src/main/java/com/powsybl/openloadflow/network/impl/PropagatedContingency.java:463]: Yes, exactly. Note that this method `apply` is not called always in some DC sensitivity analyses, that is why we have a lot of treatments in the `PropagatedContingency` class.

**Comment by annetill** [comment]
Not really easy to add a unit test, no clear issue (throw for example) in real cases I tested since yesterday.

**Review by annetill — APPROVED** [review]
Approved

**Review by annetill — APPROVED** [review]

**Pull request merged**